src/routes/blog/post/state-of-natural-language-processing/+page.markdoc

```diff
@@ -10,3 +10,152 @@ category: product
+
+Fast forward to 1966, and we have ELIZA, a program that simulated conversation by matching patterns in user input. ELIZA showed that machines could interact in a way that mimics human conversation, much like modern chatbots.
+
+Here's an example of what a chat with ELIZA might look like:
+
+```
+User: I'm feeling down today.
+ELIZA: I'm sorry to hear that. Can you tell me more about what's bothering you?
+User: I had a fight with my best friend.
+ELIZA: Fights with close friends can be tough. How did that make you feel?
+```
+
+ELIZA's responses aren't perfect, but they demonstrate the basic principle of pattern matching that laid the groundwork for more advanced NLP techniques.
+
+By the end of the 20th century, the use of statistical methods began to change NLP. Instead of using fixed rules, these methods allowed computers to learn from data. This was a big step forward and set the stage for the use of machine learning in NLP. The 2000s saw further advancements with algorithms that could learn from vast amounts of data, leading to significant improvements in tasks like language translation and speech recognition.
+
+The development of neural networks, especially word embeddings like [Word2Vec](https://arxiv.org/abs/1301.3781), marked another leap forward. These techniques allowed for more nuanced understanding of language by representing words in a space where the distance between words captured their semantic similarity. 
+
+Picture a simple 2D word embedding space:
+
+```
+      +------------+
+      |  cat       |
+      |     dog    |
+      |            |
++-----+------------+-----+
+|     |            |     |
+|     |            |     |
+|     |            |     |
++-----+------------+-----+
+      |            |
+      |  car       |
+      |     bike   |
+      +------------+
+```
+
+In this space, similar words like "cat" and "dog" are closer together, while unrelated words like "car" and "bike" are farther apart. This is a simplified view, but it illustrates how word embeddings capture semantic relationships.
+
+The introduction of transformer models in 2017, such as BERT and GPT, using [attention mechanisms](https://arxiv.org/abs/1706.03762), was another major milestone. These models could handle long pieces of text more effectively, leading to better performance on a wide range of NLP tasks.
+
+# Traditional NLP models and embeddings
+
+Before diving into the intricacies of LLMs, let's take a step back and look at some traditional NLP techniques that paved the way for these advanced models.
+
+## Naive Bayes classifiers
+
+Consider building a spam email detector. You could use a Naive Bayes classifier, which learns from examples of spam and non-spam emails to predict whether a new email is spam. 
```

Comment:
attach a link for naive bayes

Reply:
No original paper, attached this nice IBM lnk